src/vs/workbench/services/lifecycle/electron-sandbox/lifecycleService.ts

```diff
@@ -180,6 +185,7 @@ export class NativeLifecycleService extends AbstractLifecycleService {
 
 		try {
 			await raceCancellation(Promises.settled(joiners), cts.token);
+			await raceCancellation(Promises.settled(lastJoiners.map(lastJoiner => lastJoiner())), cts.token);
```

Comment:
This change means that an rejection thrown in the Default priority joiners will cause the last joiners to not run. This should be pretty rare but we should handle this case